server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -63,6 +63,8 @@
     private final Set<String> consumedParams = new HashSet<>();
     private final SetOnce<XContentType> xContentType = new SetOnce<>();
 
+    private Exception exception;
```

Comment:
Adding this mutable field (and thus making `RestRequest` not immutable) is not attractive.

Reply:
@jasontedor thanks!
 Have another look ?